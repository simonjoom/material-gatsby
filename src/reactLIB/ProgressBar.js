import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const ProgressBar = ({ progress, className, variant, valueBuffer, color }) => {
  let style;
  const classes = {
    indeterminate: progress === undefined,
    determinate: !!progress
  };
  if (progress) {
    style = {
      width: progress + '%'
    };
  }
  const inlineStyles = { bar2: {} };
  const dashedClass = cx('dashed', {
    dashedColorPrimary: color === 'primary',
    dashedColorSecondary: color === 'secondary'
  });
  const bar2ClassName = cx('bar', {
    barcolorPrimary: color === 'primary' && variant === 'buffer',
    barcolorSecondary: color === 'secondary' && variant === 'buffer',
    bar2Buffer: variant === 'buffer'
  });
  if (variant === 'buffer') {
    if (valueBuffer !== undefined) {
      inlineStyles.bar2.transform = `scaleX(${(valueBuffer || 0) / 100})`;
    }
  }

  return (
    <div
      className={cx('progress lighten-5', className)}
      style={variant === 'buffer' ? { backgroundColor: 'transparent' } : {}}
    >
      {variant === 'buffer' ? <div className={dashedClass} /> : null}
      <div className={cx(classes, className)} style={style} />
      {variant === 'buffer' ? (
        <div className={bar2ClassName} style={inlineStyles.bar2} />
      ) : null}
    </div>
  );
};

ProgressBar.defaultProps = {
  color: 'primary',
  variant: 'indeterminate'
};
ProgressBar.propTypes = {
  className: PropTypes.string,
  /**
   * A number between 0..100 that indicates the current progress,
   * when provided, a determinate bar with the progress is displayed,
   * otherwise a indeterminate bar is shown
   */
  progress: PropTypes.number
};

export default ProgressBar;
