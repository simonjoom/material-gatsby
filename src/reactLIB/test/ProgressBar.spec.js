import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from '../src/ProgressBar';

let wrapper = shallow(<ProgressBar />);

describe('<ProgressBar />', () => {
  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders by default with an indeterminate state`', () => {
    expect(
      wrapper
        .find('.progress')
        .childAt(0)
        .find('.indeterminate')
    ).toHaveLength(1);
  });

  test('handles progress prop', () => {
    wrapper = shallow(<ProgressBar progress={50} />);
    expect(
      wrapper.contains(<div className="determinate" style={{ width: '50%' }} />)
    ).toEqual(true);
  });

  test('handles different colors', () => {
    wrapper = shallow(<ProgressBar className="red" progress={0} />);
    expect(wrapper.containsMatchingElement(<div className="red" />)).toEqual(
      true
    );
  });
});
