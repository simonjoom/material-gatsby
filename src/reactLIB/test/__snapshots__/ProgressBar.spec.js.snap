// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders 1`] = `
<div
  className="progress lighten-5"
>
  <div
    className="indeterminate"
  />
</div>
`;
